File: exercises/java-core/12-exception/ExceptionDemo.java
/**
 * 第 12 章练习：异常处理
 * 对照文件：exception-demo.js
 *
 * 练习要求：
 * 1. 写 divide(int a, int b)：b 为 0 时抛 IllegalArgumentException（带提示信息）；
 *    两个调用方分别用 try-catch 就地处理 和 throws 继续上交，体会两条路线
 * 2. 自定义 InsufficientBalanceException（继承 RuntimeException，带错误码 1001），
 *    写 withdraw(double balance, double amount) 在余额不足时抛出并捕获打印
 * 3. 用 catch 打印 e.getClass().getName()，观察 Files.readString 读不存在文件时
 *    抛的异常类型，说出它属于受检还是非受检，为什么
 */
public class ExceptionDemo {

    public static void main(String[] args) {
        // 在这里写你的练习代码
    }
}

File: exercises/java-core/12-exception/exception-demo.js
/**
 * 第 12 章练习：异常处理（JS 对照）
 * 对照文件：ExceptionDemo.java
 *
 * 练习要求：
 * 1. JS 版 divide：throw new Error('...')——注意 JS 可以 throw 任何值（甚至字符串），Java 只能抛 Throwable 子类
 * 2. 自定义 class InsufficientBalanceError extends Error，实现同样的取款逻辑
 * 3. 关键思考：JS 没有"受检异常"——函数可能抛什么全靠文档和运气，
 *    Java 的 throws 把"可能失败"写进了方法签名，你怎么看这个设计？
 */

// 在这里写你的 JS 实现
